nsv13/code/modules/overmap/weapons/plasma_gun.dm

```diff
@@ -0,0 +1,466 @@
+		say("DANGER! Not enough phoron to safely dischard core! Please ensure enough gas is present before firing!")
+		return FALSE
+	if(loader.on)
+		say("DANGER! Phoron Gas Regulator back pressure surge avoided! Ensure the regulator is off before operating!")
+		return FALSE
+	if(alignment < 90)
+		if(prob(10))
+			misfire()
+			return FALSE
+	if(alignment < 75)
+		if(prob(25))
+			misfire()
+			return FALSE
+	if(alignment < 50)
+		if(prob(50))
+			misfire()
+			return FALSE
+	if(alignment < 25)
+		misfire()
+		return FALSE
+	else
+		return TRUE
+
+/obj/machinery/ship_weapon/plasma_caster/local_fire()
+	for(var/mob/living/M in get_hearers_in_view(7, src)) //burn out eyes in view
+		if(M.stat != DEAD && M.get_eye_protection() < 2) //checks for eye protec
+			M.flash_act(10)
+			to_chat(M, "<span class='warning'>Your eyes burn from the brilliance of the Lämp!</span>")
+
+/obj/machinery/ship_weapon/plasma_caster/do_animation()
+	set waitfor = FALSE
+	sleep(2 SECONDS)
+	flick("[initial(icon_state)]_firing",src)
+	sleep(fire_animation_length)
+	icon_state = initial(icon_state)
+
+/obj/machinery/ship_weapon/plasma_caster/animate_projectile(atom/target)
+	return linked.fire_projectile(weapon_type.default_projectile_type, target, homing = TRUE, speed = 2, lateral=weapon_type.lateral)
+
+/obj/machinery/ship_weapon/plasma_caster/proc/misfire()
+	say("WARNING! Phoron containment field failure, ejecting gas!")
+	if(prob(15))
+		do_sparks(4, FALSE, src)
+	if(plasma_mole_amount > 0 && prob(5))
+		makedarkpurpleslime()
+	atmos_spawn_air("plasma=[plasma_mole_amount];TEMP=293")
+	alignment -= rand(30,90)
+	if(alignment < 0)
+		alignment = 0
+	field_integrity -= rand(20,50)
+	plasma_mole_amount = 0
+	playsound(src, malfunction_sound, 100, 1)
+	tesla_zap(src, 8, TESLA_MINI_POWER, TESLA_ENERGY_MINI_BALL_FLAGS)
+	playsound(src,'sound/machines/defib_zap.ogg', 100, 1)
+	unchamber()
+	unload()
+
+/obj/machinery/ship_weapon/plasma_caster/overmap_fire(atom/target)
+
+	if(weapon_type?.overmap_firing_sounds)
+		overmap_sound()
+
+	if(overlay)
+		overlay.do_animation()
+	sleep(4 SECONDS)
+	if( weapon_type )
+		animate_projectile(target)
+
+/obj/machinery/ship_weapon/plasma_caster/after_fire()
+	alignment -= rand(30,60)
+	if(alignment < 0)
+		alignment = 0
+	plasma_mole_amount = 0
+	field_integrity -= 20
+	cooldown = 100
+	light_power = 2
+	..()
+
+/obj/machinery/ship_weapon/plasma_caster/default_deconstruction_crowbar(obj/item/I, ignore_panel)
+	var/mob/living/fool = usr
+	var/confirm = alert("You feel an electric tingle as you bring your crowbar close, there's an odd heat the closer you get.\
+						 Are you sure this is a good idea?", "Deconstruct Phoron Caster", "Yes", "No")
+	if(confirm == "Yes")
+		visible_message("<span class='danger'>Burning energy and phoron starts to vent from the gun which chars [usr]!</span>")
+		field_integrity -= rand(30,50)
+		alignment -= rand(50,80)
+		if(alignment < 0)
+			alignment = 0
```

Comment:
```suggestion
		alignment = max(alignment - rand(50,80), 0)
```
This is cleaner, max returns the highest value of the two, so if `alignment - rand(50,80)` is smaller than 0 it will return 0